### PR TITLE
mesos-dns bugfixes and truncate bit disable

### PR DESCRIPTION
**Pull request by alberts**
## High Level Description

Update mesos-dns to build instead of downloading a binary.

Fix race between mesos-dns serving empty data (NXDOMAINs) and polling state.json at startup.

Updates mesos-dns to HEAD, bringing in a couple of commits from the last few months.

Also included code that disables setting the truncate bit and config change to activate it.

## Related Issues

  - [DCOS-10809](https://mesosphere.atlassian.net/browse/DCOS-10809): DNS unavailable during DC/OS 1.8.5 upgrade
 - [DCOS-13193](https://mesosphere.atlassian.net/browse/DCOS-13193): Review EDNS fixes, merge to master if needed
 - [DCOS-13507](https://mesosphere.atlassian.net/browse/DCOS-13507): build mesos-dns in DC/OS build instead of pulling from CI

## Checklist for all PR's

  - [x] Included a test which will fail if code is reverted but test is not.
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated: [change log](https://github.com/mesosphere/mesos-dns/compare/43cd0620400449b70ee52883e94ec1c38f84b1ac...c8eb70e77f1f0ee8fa590421a89a665abc726192)
  - [x] Test Results: https://jenkins.mesosphere.com/service/jenkins/job/public-mesos-dns-pulls/159/
  - [x] Code Coverage https://jenkins.mesosphere.com/service/jenkins/job/public-mesos-dns-pulls/159/

### CODE REVIEW

File: packages/mesos-dns/buildinfo.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-  "sources": {
-    "mesos-dns": {
-      "kind": "url",
-      "url": "https://github.com/mesosphere/mesos-dns/releases/download/v0.6.0/mesos-dns-v0.6.0-linux-amd64",
-      "sha1": "da98d671ca743ab44aa876b65d4e08db22b8c4d3"
-    }
+  "docker": "golang:1.7",
+  "single_source" : {
+    "kind": "git",
+    "git": "git@github.com:mesosphere/mesos-dns.git",
```
Comment (spahl) [inline packages/mesos-dns/buildinfo.json]: this should be https

File: packages/mesos-dns/buildinfo.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {
-  "sources": {
-    "mesos-dns": {
-      "kind": "url",
-      "url": "https://github.com/mesosphere/mesos-dns/releases/download/v0.6.0/mesos-dns-v0.6.0-linux-amd64",
-      "sha1": "da98d671ca743ab44aa876b65d4e08db22b8c4d3"
-    }
+  "docker": "golang:1.7",
```
Comment (nlsun) [inline packages/mesos-dns/buildinfo.json]: We probably shouldn't use 1.7 here until the mesos-dns CI uses 1.7

Comment (alberts) [inline packages/mesos-dns/buildinfo.json]: @nlsun Time to update CI. :)

Comment (nlsun) [inline packages/mesos-dns/buildinfo.json]: I've update mesos-dns CI to use 1.7.5 and tests are still passing.

File: packages/mesos-dns/build [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
+
+export GOPATH=/pkg:$GOPATH
+export PATH=/pkg/bin:$PATH
+
+go get github.com/tools/godep
+
+mkdir -p /pkg/src/github.com/mesosphere
+mv /pkg/src/mesos-dns /pkg/src/github.com/mesosphere/
+cd /pkg/src/github.com/mesosphere/mesos-dns
+godep go install -v ./...
```
Comment (nlsun) [inline packages/mesos-dns/build]: we actually don't need this or the above line installing godep

File: packages/mesos-dns/buildinfo.json (hunk 2 above)
Comment (alberts) [inline packages/mesos-dns/buildinfo.json]: awesome

File: packages/mesos-dns/buildinfo.json [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
+  "single_source" : {
```
Comment (gpaul) [inline packages/mesos-dns/buildinfo.json]: Is this rename intentional?

Comment (alberts) [inline packages/mesos-dns/buildinfo.json]: indeed it is

**Comment by orsenthil** [comment]
Is there any corresponding enterprise change need for this PR? If yes, is there an enterprise PR?

**Comment by gpaul** [comment]
@orsenthil great catch. 

Here you go, it should be ready to board
https://github.com/mesosphere/dcos-enterprise/pull/558
